nucypher/blockchain/eth/signers.py

```
@@ -191,3 +216,93 @@ def unlock_account(self, account: str, password: str, duration: int = None) -> b
+        return Web3.isAddress(account) and \
+            self.account == Web3.toChecksumAddress(account)
+
+    @classmethod
+    def from_signer_uri(cls, uri: str) -> 'Signer':
+        return cls(uri)
+
+    @property
+    def account(self) -> str:
+        return self._account
+
+    @validate_checksum_address
+    def is_device(self, account: str) -> bool:
+        return False
+
+    def accounts(self) -> List[str]:
+        return [self.account]
+
+    @validate_checksum_address
+    def unlock_account(self, account: str, password: str, duration: int = None) -> bytes:
+        if not self._is_my_account(account):
+            return False
+
+        if self._signer is None:
+            try:
+                self._signer = Account.from_key(
+                    Account.decrypt(self.key, password)
+                )
+            except ValueError:
+                return False
+
+            if self.account != self._signer.address:
+                raise ValueError("unexpected address '%s' recovered from private key; expected '%s'" % (
+                    self._signer.address,
+                    self.account,
+                ))
+
+        return True
+
+    @validate_checksum_address
+    def lock_account(self, account: str) -> str:
+        if not self._is_my_account(account):
+            return False
+
+        self._signer = None
+        return True
+
+    @validate_checksum_address
+    def sign_transaction(self, transaction_dict: dict) -> HexBytes:
+        # Do not include a 'to' field for contract creation.
+        if not transaction_dict['to']:
+            transaction_dict = dissoc(transaction_dict, 'to')
+
+        return self._signer.sign_transaction(
+                transaction=transaction_dict,
+            ).raw
+
+    @validate_checksum_address
+    def sign_message(self, account: str, message: bytes, **kwargs) -> HexBytes:
+        if not self._is_my_account(account):
+            return None
```

This is a very odd scenario (current design expects multi-account usage)... Let's raise an exception instead.

---

Done - sign_message() now throws if the account is unexpected.